src/metabase/lib/drill_thru/underlying_records.cljc

```diff
@@ -92,8 +115,20 @@
      :table-name (when-let [table-or-card (or (some->> query lib.util/source-table-id (lib.metadata/table query))
                                               (some->> query lib.util/source-card-id  (lib.metadata/card  query)))]
                    (lib.metadata.calculation/display-name query stage-number table-or-card))
-     :dimensions dimensions
-     :column-ref column-ref}))
+     ;; If no dimensions were provided but the underlying column comes from an aggregation, then construct the
+     ;; dimensions from the row data.
+     :dimensions (or (not-empty dimensions)
+                     (when (and (not (aggregation-sourced? column))
+                                (aggregation-sourced? query column))
+                       (into [] (filter #(breakout-sourced? query (:column %))
+                                        row))))
```

Comment:
```suggestion
                       (filterv #(breakout-sourced? query (:column %)) row)))
```

Reply:
Should this be wrapped in `not-empty`?

Reply:
You know, I initially had `(seq row)` in the `when` condition guarding this and then removed it, but on second thought I agree `dimensions` should be `nil` if `row` is empty.